### ISSUE TITLE
♻️ Upgrade to Laravel 11

### ISSUE THREAD

**Issue by petertgiles** [GCTC-NTGC/gc-digital-talent#9754]
## ♻️ Debt/Refactor

Laravel 11 is out!
https://laravel-news.com/laravel-11

## 🙋‍♀️ Proposed Solution

https://laravel.com/docs/11.x/upgrade#upgrade-11.0

## ✅ Acceptance Criteria

- [ ] API is updated to version 11

## 🛑 Blockers

```[tasklist]
### Blocked By
- [ ] https://github.com/GCTC-NTGC/gc-digital-talent/issues/10187
- [ ] https://github.com/GCTC-NTGC/gc-digital-talent/issues/11310
```

**Comment by mnigh** [comment]
It looks like this could be a bit of a pain: https://laravel.com/docs/11.x/upgrade#modifying-columns

**Comment by vd1992** [comment]
Yeah, per the document we will likely need migration squashing if not updating every instance of `change()`

**Comment by vd1992** [comment]
Test database restore before going down the squash path
Consider splitting squashing into its own issue

**Comment by petertgiles** [comment]
Looks like we meet the curl 7.34.0 requirement, at least in the local container.
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/088b7976-6239-4682-9446-bb60d868bded)

**Comment by vd1992** [comment]
Updated blockers

**Comment by mnigh** [comment]
> Looks like we meet the curl 7.34.0 requirement, at least in the local container. ![image](https://private-user-images.githubusercontent.com/8978655/325696170-088b7976-6239-4682-9446-bb60d868bded.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQwNjQ4MjMsIm5iZiI6MTcxNDA2NDUyMywicGF0aCI6Ii84OTc4NjU1LzMyNTY5NjE3MC0wODhiNzk3Ni02MjM5LTQ2ODItOTQ0Ni1iYjYwZDg2OGJkZWQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MjVUMTcwMjAzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGE2OWNlYTA1OGQzMmNiNTM5ZDk5YmRmMTg1ZWM3MzZjY2JmODk4YTMyZDI4Nzg0ZjZiYjIyOTI5MjcxMWVkMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.yXHbADgnXpGa-4UjfBcAGBVkW3dKfMJdVKadWFSEyPY)

Let us celebrate that we are not using a version of cURL that is [over a decade old](https://curl.se/docs/releases.html#:~:text=166,85) 🥳

**Comment by tristan-orourke** [comment]
@mnigh We've squashed the migrations and tested the database backup. Is there anything else blocking this now?

**Comment by petertgiles** [comment]
Currently blocked by the scout plugin:
https://github.com/devNoiseConsulting/laravel-scout-postgres-tsvector/issues/5

**Comment by esizer** [comment]
Maybe time to make a decision on that dep that is blocking us @tristan-orourke ? 😅

 I believe @petertgiles suggested we just use database triggers and avoid scout for keeping the index up to date?

**Comment by tristan-orourke** [comment]
How much work would it take to move away from Scout?

Honestly, my instinct is to fork the repo we're using. Seems like upgrading it for Laravel 11 is trivial - it seems like someone else has already forked it and done just that: 
https://github.com/mobilozophy/laravel-scout-postgres-tsvector/commit/a1dd7c9b73a9aefffb57d7bd3754fb7857c3216e
